From b2b9c8c2277ec1754d49872c1427ef41b75385f8 Mon Sep 17 00:00:00 2001
From: Han Xu <b45815@freescale.com>
Date: Thu, 11 Jun 2015 14:50:47 -0500
Subject: [PATCH 0742/5242] MLK-11095: mtd:qspi: clear the DDR_EN bit on 6UL
 and 7D

commit  cc66c86576953571b7dae0daf62aa22b2b6f745c from
https://source.codeaurora.org/external/imx/linux-imx.git

the obsolete bit DDR_EN on 6UL and 7D should be clear in case other
program set the bit and cause qspi probe fail.

Signed-off-by: Han Xu <b45815@freescale.com>
(cherry picked from commit d8b51cc358780f68e732522ee9bd6bd578dd6771)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mtd/spi-nor/fsl-quadspi.c |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/mtd/spi-nor/fsl-quadspi.c b/drivers/mtd/spi-nor/fsl-quadspi.c
index 1ff3430..3d0e480 100644
--- a/drivers/mtd/spi-nor/fsl-quadspi.c
+++ b/drivers/mtd/spi-nor/fsl-quadspi.c
@@ -760,6 +760,14 @@ static int fsl_qspi_nor_setup(struct fsl_qspi *q)
 	if (ret)
 		return ret;
 
+	if ((q->devtype_data->devtype == FSL_QUADSPI_IMX6UL) ||
+		(q->devtype_data->devtype == FSL_QUADSPI_IMX7D)) {
+		/* clear the DDR_EN bit for 6UL and 7D */
+		reg = readl(base + QUADSPI_MCR);
+		writel(~(QUADSPI_MCR_DDR_EN_MASK) & reg, base + QUADSPI_MCR);
+		udelay(1);
+	}
+
 	/* Reset the module */
 	qspi_writel(q, QUADSPI_MCR_SWRSTSD_MASK | QUADSPI_MCR_SWRSTHD_MASK,
 		base + QUADSPI_MCR);
-- 
1.7.9.5

